Unit Tests for events.py

deseansoh opened the pull request:
Unit tests to check that functions in events.py are working as intended. Some things to note:
- check_for_wiegand is not being used anywhere in the codebase. Mark for deletion?
- reader_detects_bits is a very nested and dependent function, so currently UTs have not been written for it yet
- check_for_masterpassword and verify_authtype aren't being used in the codebase as well. I'm guessing they were created to be used in reader_detects_bits but they aren't being used so do we incorporate them in or delete them?
- verify_zone_status is used in update_zone_status, which aren't being used anywhere else in the codebase as well
- verify_antipassback is also not being used anywhere else
Seems like all the verify_ functions were meant to be used inside reader_detects_bits, but currently aren't

deseansoh committed:
Clarification for program.py

deseansoh committed:
Clarification for events.py

deseansoh committed:
Clarification for eventActionTriggerConstants.py

deseansoh committed:
Clarification for eventsMod.py

deseansoh committed:
Clarification for eventsMod.py

deseansoh committed:
Update events.py

deseansoh committed:
Clarification for relay.py

deseansoh committed:
Clarification for server_init.py

deseansoh committed:
Clarification for api.py

deseansoh committed:
Clarification for relay.py

deseansoh committed:
Renaming testing folder to module test

deseansoh committed:
Creating test_example for testing on pi

deseansoh committed:
Testing imports

deseansoh committed:
Revert imports

deseansoh committed:
Import testing

deseansoh committed:
Adding relative path

deseansoh committed:
Moving test folder to src

deseansoh committed:
Editing relative path

deseansoh committed:
Update test_example.py

deseansoh committed:
Update test_example.py

deseansoh committed:
Moving out test folder, import settings

deseansoh committed:
Trying if tests work in src folder

deseansoh committed:
Delete test_example_in_src.py

deseansoh committed:
Create test_example_in_src.py

deseansoh committed:
Refactor logs path

From hardcoded home/etlas to be based on home directory instead

deseansoh committed:
Importing OS for files without

deseansoh committed:
Delete test_example_in_src.py

deseansoh committed:
Create test_example_in_src.py

deseansoh committed:
Updating with failing test to check working

deseansoh committed:
Reset output_events list after each test

deseansoh committed:
Experimenting with test in src

deseansoh committed:
Test initiator, starting UT for eventActionTriggers

deseansoh committed:
Fail test case for update EAT

deseansoh committed:
Function name error fix

deseansoh committed:
Documentation for first 2 functions

deseansoh committed:
check_datetime tests

deseansoh committed:
get_entrance_from_ev_mngment tests

deseansoh committed:
Experimenting event_trigger_cb

deseansoh committed:
Correcting mock json

deseansoh committed:
Adding json fixture, experimenting event trig cb 2

deseansoh committed:
Reformat errors

deseansoh committed:
Correct str concat error

deseansoh committed:
event trigger cb Experiementation 3

deseansoh committed:
json edits

deseansoh committed:
json correction

deseansoh committed:
Adding additional event trigger cb test

deseansoh committed:
Function name change

deseansoh committed:
Experimenting event trigger cb 4

deseansoh committed:
event trigger cb experimentation 5

deseansoh committed:
event trigger cb experimentation 6

deseansoh committed:
event trigger cb experimentation 7

deseansoh committed:
event action trigger experimentation 8

deseansoh committed:
event trigger cb experimentation 9

deseansoh committed:
event trigger cb experimentation 10

deseansoh committed:
mock json edits

deseansoh committed:
update mock json errors

deseansoh committed:
event trigger cb test Code Cleanup

deseansoh committed:
code cleanup 2

deseansoh committed:
Test for flush_output

deseansoh committed:
code error edits to test flush output

deseansoh committed:
flush output variable scoping fix

deseansoh committed:
variable scoping fix 2

deseansoh committed:
Cleaning up code for eventActionTriggers Testing

deseansoh committed:
Creat test_events

deseansoh committed:
Test for verify_datetimes

deseansoh committed:
Experiment verify datetime 1

deseansoh committed:
Experiment verify datetime 2

deseansoh committed:
Test for not verify datetime

deseansoh committed:
Test for check_entrance_status

deseansoh committed:
Experiment mocking lock_unlock_entrancex 1

deseansoh committed:
Experiment mocking lock unlock 2

deseansoh committed:
Experiment mocking lock unlock 3

deseansoh committed:
Experiment mocking lock unlock 4

deseansoh committed:
Experiment mocking lock unlock 5

deseansoh committed:
Test for check_entrance_status only E1

deseansoh committed:
Test for check entrance status both closed

deseansoh committed:
Test for open door E1

deseansoh committed:
Test for open door E2

deseansoh committed:
Test for open door both

deseansoh committed:
Test for open entrance id 1

deseansoh committed:
Test for open entrance id 2

deseansoh committed:
Code Cleanup and adding Todo

deseansoh committed:
Test for check for masterpassword

deseansoh committed:
Edit typeerror

deseansoh committed:
Combining test case for using entrance id

deseansoh committed:
Combining test case for open door

deseansoh committed:
Combining test case for check entrance status

deseansoh committed:
Code cleanup

deseansoh committed:
Test for verify authtype, fixture json file

deseansoh committed:
Updating TODO

deseansoh committed:
Test for mag_detects_rising

deseansoh committed:
Touchup Code

deseansoh committed:
Test for mag_detects_falling

deseansoh committed:
Test correction

deseansoh committed:
Testing why actual func is called

deseansoh committed:
Test correction

deseansoh committed:
Test button_detects_change

junnengsoo commented on src/api.py:
```diff
@@ -223,6 +223,7 @@ def get_piProperty():
     data = piProperty.get_system_stats()
     return flask.Response(json.dumps(data), headers={ 'Content-type': 'application/json' }, status=200)
 
+# does this only trigger button press for E1 but not E2 - DS
```
Go ahead and delete this.

deseansoh replied:
Will do

junnengsoo commented on src/eventActionTriggerConstants.py:
```diff
@@ -118,7 +118,7 @@ def get_event_entrance(event):
     return event[-1]
 
-
+# can standardise with above by renaming to get_event_event_action_trigger
```
Go ahead.

deseansoh replied:
Will do

junnengsoo commented on src/events.py:
```diff
@@ -56,7 +58,7 @@
     """A custom exception used to report errors in use of Timer class"""
 
-
+# Timer class is same as the one in timer.py, should import from there instead? - DS
```
Go ahead.

deseansoh replied:
Will do

junnengsoo commented on src/events.py:
```diff
@@ -249,6 +251,7 @@ def update_credOccur():
 
 MAX_PIN_LENGTH = 6
 
+# the vars in 253 to 261 seem to be not initialised? I'm not sure where they are populated - DS
```
Remove, since it is currently not being used, remove `check_for_wiegand` as well.

deseansoh replied:
Removed `check_for_wiegand()`, realised the vars are used within `reader_detects_bits()`, so I left them alone

junnengsoo commented on src/events.py:
```diff
@@ -284,7 +287,7 @@ def check_for_wiegand(value):
                         return {"Name": personName, "diffpassword": diffpassword, "AccessGroup": groupName, "Schedule": groupdetails["Schedule"]}
 
-
+# seems to be a helper function for open_door_using_entrance_id, should be a local function within it? - DS
```
Yes, move this to `open_door_using_entrance_id`.

deseansoh replied:
Decided to keep the function as a script-wide, and removed the local `open_door()` within `reader_detects_bits` to use this one instead

junnengsoo commented on src/eventsMod.py:
```diff
@@ -47,7 +49,7 @@ def update_config():
     MAX_JSON_LENGTH = int(config.get("archivedMAXlength", 10))
 
-
+# isn't this handled in line 48
```
Remove this, set `MAX_JSON_LENGTH = DEFAULT_JSON_LENGTH`. `DEFAULT_JSON_LENGTH = 10`.

deseansoh replied:
Will do

junnengsoo commented on src/eventsMod.py:
```diff
@@ -226,7 +228,7 @@ def record_antipassback(authtype, entrance, status):
         "eventTime": datetime.now().strftime(("%m-%d-%Y %H:%M:%S"))
     }
-
+    # Top dictionary is overwritten so no need for it?
```
Okay.

deseansoh replied:
Will do

junnengsoo commented on src/healthcheck.py:
```diff
@@ -98,6 +98,7 @@ def get_host_ip(hostIP=None):
                     time.sleep(0.1)
 
         if str(hostIP).startswith('169.254') and (not check_ip_static()):  # apipa, use static ip
+            # just to check, why is the changing of static IP to 230, and not to 160 or 250? - DS
```
This is for the initial setup to allow for default ip assignment to new connecting controllers. This also allows for easier SSH due to the default being 230 for new controllers, thereafter then we change the ip address. New controllers then can be connected with the default ip address, and the process repeats.

deseansoh replied:
Okay will remove the clarification comment

junnengsoo commented on src/program.py:
```diff
@@ -95,6 +95,7 @@ def check_events_for(entrance):
 
     if timeout_buzzer.status():
         if timeout_buzzer.check(BUZZER_TIMEOUT):
+            # Should this be changed to API call to backend? - DS
```
Raise an issue for this. We will implement timeout feature of door left opened.

deseansoh replied:
Will do

junnengsoo commented on src/program.py:
```diff
@@ -133,6 +134,7 @@ def check_entrance_E1():
 
 
 def check_entrance_E2():
+    # should use relay.lock_unlock_entrance_XXX() instead? - DS
```
TBH this is not even used right now, remove.

deseansoh replied:
Will do

junnengsoo commented on src/relay.py:
```diff
@@ -99,7 +101,7 @@ def setRelayPinLow(relayPin):
 # *** Relay activate/deactivate/toggle ***
 
-
+# could possible combine the bottom two functions into one? and have activate level determine whether to set high or low? - DS
```
Yes. Do that.

deseansoh replied:
I have done this in the `UnitTest-relay` branch, and tested it on the Unicon as well to ensure that it works fine. It can be merged into `develop` branch once it has been reviewed

junnengsoo commented on src/relay.py:
```diff
@@ -126,6 +128,8 @@ def toggleRelay1(relayPin, activateLevel, activateMilliSeconds, deActivateMilliS
         setupRelayPin(relayPin)
 
+        # Unsure about logic here, it is supposed to activate and deactivate toggleCount times? If so, seems like the
+        # code from line 141 should be in the for loop?
```
Currently I dont see why we need `toggleCount` lets discuss this with Yong Ning

deseansoh replied:
Discussed this with YN while doing up the unit tests in the `UnitTest-relay` branch, and concluded that there is no need for toggleCount currently, so I have removed it as well in the changes I made in that branch.

junnengsoo commented on src/server_init.py:
```diff
@@ -106,7 +106,7 @@ def run_servers():
 #print ("get_netmask():" + get_netmask())
     
-
+# maybe can change the name of this function to something better? like set_connection() - DS
```
Yup sure, infact that are multiple other `main()` methods in the codebase that we can rename

deseansoh replied:
Will do

junnengsoo commented on test/unit/test_eventActionTriggers.py:
```diff
@@ -0,0 +1,294 @@
+
+    assert EAT.get_entrance_from_event_management(mock_json_contents) == EAT.BOTH_ENTRANCE
+
+def test_event_trigger_cb_1(monkeypatch: pytest.MonkeyPatch, mock_json_eventActionTriggers):
```
Let's rewrite the actual `event_trigger_cb()` function for readability.

deseansoh committed:
Remove exit button api function

deseansoh committed:
Rename&refactor get_event_trigger_from_event

Changed to get_event_event_action_trigger, to standardise with above

deseansoh committed:
Delete timer.py

Only one is needed, and it is already declared as a local class within events.py, which is the only place it is used

deseansoh committed:
Remove additional comment

deseansoh committed:
Remove check_for_wiegand

deseansoh committed:
Removed local open_door in reader_detects_bits

Removed the local function within reader_detects bits to use the one defined in the class instead

deseansoh committed:
Editing and fixing MAX_JSON_LENGTH

deseansoh committed:
Remove redeclared dict in record_antipassback

deseansoh committed:
Remove clarification comment

deseansoh committed:
Remove clarification comment for timeout_buzzer

deseansoh committed:
Remove check_entrance_E1 and E2

Removed these two functions within program.py because they are not being used

deseansoh committed:
Change main to set_connection
